### DISCUSSION TITLE
LTSP-manager is it needed and what should it provide

### DISCUSSION THREAD

**Discussion by enaut** [ltsp/ltsp#199] [Q&A]
LTSP-manager was discontinued. Some questions were asked about it in ltsp/ltsp#187 and ltsp/ltsp#193. I would continue development if there is need for it.  However I'm not really sure what it should look like in future so I'd like some feedback from users. 

Features that were available in the past:

* File:
  * Allow user signups (users can register fromthe login screen and need to be confirmed by the admin)
  * Batch create users
  * import from passwd, csv
  * export to csv
* Server (ltsp specific ):
  * initial setup
  * network config
  * update image
  * revert image
  * edit config files (lts.conf, dnsmasq.conf, pxelinux.conf)
* View:
  * ...
* Groups:
  * Add, delete and edit groups
* Users:
  * Add delete and edit users
  * remove user from group

My questions are:
* What functionality does it need?
* What can be dropped?
* What should be implemented?
* Who would use it?
* What are the key features?
* Would/Should it be useful without ltsp?
* Anything I forgot?

**Comment by enaut** [comment]
### Done so far:
* fixed bugs
* progressbars for everything taking time
* ported to python3
* dev: ported to meson
* recently moved to github: https://github.com/enaut/ltsp-manager/
* created a personal ppa:
  * ppa:franz-die/ltsp-manager
  * the most recent version is at: ppa:franz-die/ltsp-manager-testing

### Ideas so far:
* rewrite so that the GUI runs without root (prooved to be quite hard.)
  * needs sepparation in gui and dbus daemon
  * would be mostly easy if https://gitlab.freedesktop.org/accountsservice/accountsservice/issues/36 was included.
* reimplement the ltsp specific stuff for ltsp19?

So @dgroos and @rkwesk and all others, what do you need ltsp-manager for?

**Comment by rkwesk** [comment]
My experience with ltsp has mostly been helping to setup and maintain it (now that I'm retired) in state (public) schools in Greece with students from 6 to 19 years old. Alkis Georgopoulos had created a particular version of the ltsp manager called sch-scripts which had additional features particularly targeted for schools in Greece. When one installed sch-scripts it also installed all the ltsp5 packages and configured epoptes and dnsmasq and setup the shared folder and other details that don't come to mind at the moment.

What was obvious was that the admin/teacher of the school lab was thus able to get a server up and running without having much knowledge and/or without using time that she/he did not have.

The result was that close to 2000 schools adopted ltsp with Alkis ready to answer questions in a forum and live on irc and even with vlc to go in to a system remotely (and he still does) and has also created a new version for ltsp19, again targeting schools in Greece.

For a short time he or his team also created ltsp manager that was to be useful to others (not Greek schools.) The interest in this app seemed not to be enough to warrant their continuing work on it.

So to your question: I think there are potentially many, not very knowledgeable or skilled and with little time to setup or maintain ltsp, people who would clearly benefit from such an app. So if it installed all the ltsp19 packages with options for headless or not servers and dnsmasq and epoptes including adding the admin user to the epoptes group. As some changes require ltsp image followed by ltsp initrd and occasionally ltsp -o ipxe and ltsp -o nfs I think it worthwhile having mini-scripts behind some buttons so that several ltsp commands would be invoked in the correct order.

One example is that if a conf file is edited it might require ltsp initrd to be invoked once the conf file is saved and since it takes so little time to run ltsp -o ipxe and ltsp -o nfs and ltsp initrd it seems harmless to have them invoked after many other changes.

Richard

**Comment by dgroos** [comment]
Hi Franz and Richard,

LTSP- Manager is a great idea -- as you have been talking about, Richard, I
think ltsp can become a more mainstream approach in so many kinds of
places, my main interest is in non-profits. Alkis and friends have brought
ltsp into a modern space with ltsp19 and the python3 version of epoptes. As
a non IT professional, I really really appreciate the possibility of being
able to get started with it w/out needing much technical knowledge and
support. Long way of saying that I think ltsp-manager could be a big part
of that vision. I expect to have more time this weekend to consider
questions of possibilities and needs. Thanks Franz for asking!

David

**Comment by dgroos** [comment]
> What was obvious was that the admin/teacher of the school lab was thus able to get a server up and running without having much knowledge and/or without using time that she/he did not have.

An important sub-set of potential ltsp fans include teachers who see the huge utility of students having access to networked computers for their learning and are not afraid of learning and troubleshooting and making tech systems work. There are many in this group who might know a little CLI. And, school and district tech leaders like this should be considered in this targeted audience of ltsp, too. 

So, to answer your question @enaut I envision LTSP-Manager as a GUI wrapper app that would walk a user such as described above through the steps of setting up an ltsp19 server. Here's a design of an idea that's obviously fragmented, pie-in-the-sky, borrowing from previous ltsp-manager and wiki pages: it's just to get a structure out there to explore, tear apart and transform. OK, to begin:

LTSP-Manager could walk the new user through 3 stages. A user could choose to skip a stage as needed or likewise to quit at any time but upon restart of the program return to that particular point.

**Stage 1 -- Pre-setup** (This would be a repeat of info that's on the ltsp site but that's important for noobies to re-read when actually making the set-up decisions.)

1. Reviews the generic structure of an ltsp system, using the chrootless approach, w/out calling it chrootless, on a desktop in a LAN (not considering details of dns/dhcp, 1 or 2 NIC).

1. Reviews the 1 NIC/ 2 NIC approaches, pros/cons of each. Links available to pages explaining basics of dns and dhcp as needed. As an aside, I think the tech-teachers wanting to use this in just their classroom will be most interested in the 2 NIC system, those who work at a higher level of the school/organization level will be more interested in the 1 NIC approach.

1. Describe briefly how this could be setup in a Virtualbox setup (w/links to detailed info) for testing or, on a desktop computer for a classroom or school-wide system implementation.

1. Describe briefly user creation approaches eg local users, LDAP users (and ?). I'm guessing that for the teacher-users LDAP is out of the question but info for future possibilities could be given. The structure of the needed csv would be given (and a user could quit ltsp-manager, make the file and come back the next day and restart here...)

**Stage 2: Actual setup** (Maybe there were check-boxes for each of the steps in Stage 1 where a person makes important choices, creating a design that's then performed at this stage...)

1. A person is walked through setting up the system that was designed (chosen) in Stage 1, including Epoptes setup and user creation.

1. Some options such as a shared folders, printer setup would be configured at this stage.

**Stage 3: Administration**

1. A person is walked through management features such as epoptes, image maintenance, etc. All details such as ipxe etc are hidden yet available to explore via links. 

1. A dashboard with various admin options are presented.

I hope that parts of this are useful for moving the discussion forward.
David

**Comment by enaut** [comment]
Thank you for the elaborate answer @dgroos! That's a really well written use case/guide for me (even though I'll not create it as you described it but there are a lot of useful hints).

Some remarks:
I guess the stage 1 Part is best done as website (as this will probably include a guide for installing). As much of it will hopefully be done by LTSP documentation I'd rather not duplicate the information - which would lead to quite some outdated docs either here or there.

I don't think ltsp-manager will go down the LDAP as LDAP has its own powerful user management.

All in all LTSP is very powerful and I'll not be able to include much of its functionality into the program. But It would probably be very good to have a few options. Right now the initial setup does help a little but without any customization.

**Comment by alkisg** [comment]
@enaut if you ever want me to create an ltsp-manager repository under https://github.com/ltsp and give you access there, just ping!

**Comment by enaut** [comment]
Thanks! - I'll do a little development and if I'm at a stage where I'm confident enough to publish it I will discuss with you what the best options might be - I also created an organization called ltsp-manager.

Progress in the last days (goal GUI and privileged service separation)

- [x] wrote dbus service for account management (probably still needs tweaking)
  - [x] prototype for privilege checking
  - [x] implement privilege checking
  - [ ] continue extending the dbus interface depending on what is needed
- [ ] make the GUI use the dbus service
  - [x] prototype use it
  - [ ] rewrite of the ltsp-manager UI - not really necessary but helps me understand.
    - [x] make ltsp-manager tabbed (like gnome-settings)
    - [x] build user list
    - [x] build groups list
    - [ ] allow editing users
      - [x] deleting user(s)
      - [x] adding users
      - [x] batch adding users
      - [ ] importing users
      - [x] editing user
    - [x] allow editing groups
      - [x] deleting groups
      - [x] adding groups
      - [x] editing groups
      - [x] adding users to group
- [ ] Long term: implement all the other privileged scripts as dbus service

Progress is good so far but now the autumn holidays are over so things will probably get slower now. 

![Bildschirmfoto von 2019-11-09 09-49-25](https://user-images.githubusercontent.com/290005/68525861-4ad5bd80-02d6-11ea-99d3-c4018073a414.png)
